src/main/java/org/maxgamer/quickshop/shop/ContainerShop.java

```diff
@@ -1160,35 +1245,93 @@ public DisplayItem getDisplayItem() {
      */
     public boolean isDoubleShop() {
         Util.ensureThread(false);
-        ContainerShop nextTo = this.getAttachedShop();
-        if (nextTo == null) {
+        if (attachedShop == null) {
             return false;
         }
-        if (nextTo.matches(this.getItem())) {
+        if (attachedShop.matches(this.getItem())) {
             // They're both trading the same item
             // They're both buying or both selling => Not a double shop,
             // just two shops.
             // One is buying, one is selling.
-            return this.getShopType() != nextTo.getShopType();
+            return this.getShopType() != attachedShop.getShopType();
         } else {
             return false;
         }
     }
 
     /**
-     * Returns the shop that shares it's inventory with this one.
-     *
-     * @return the shop that shares it's inventory with this one. Will return null if this shop is not
-     * attached to another.
+     * Updates the attachedShop variable to reflect the currently attached shop, if any.
+     * Also updates the left shop status.
      */
-    public @Nullable ContainerShop getAttachedShop() {
+    public void updateAttachedShop() {
         Util.ensureThread(false);
-        Block c = Util.getSecondHalf(PaperLib.getBlockState(this.getLocation().getBlock(), false).getState());
+        Block c = Util
+            .getSecondHalf(PaperLib.getBlockState(this.getLocation().getBlock(), false).getState());
         if (c == null) {
-            return null;
+            return;
         }
         Shop shop = plugin.getShopManager().getShop(c.getLocation());
-        return shop == null ? null : (ContainerShop) shop;
+        attachedShop = shop == null ? null : (ContainerShop) shop;
+
+        if (attachedShop != null && attachedShop.matches(this.getItem())) {
+            updateLeftShop();
+        } else {
+            attachedShop = null;
+            isLeftShop = false;
+        }
+    }
+
+    /**
+     * This function calculates which block of a double chest is the left block,
+     * relative to the direction the chest is facing. Left shops don't spawn items since
+     * they merge items with the right shop.
+     * It also updates the isLeftShop status of this class to reflect the changes.
+     */
+    private void updateLeftShop() {
+        if (attachedShop == null) {
+            return;
+        }
+
+        switch (((Chest) getLocation().getBlock().getBlockData()).getFacing()) {
```

Comment:
We should check if the shop was instanceof Chest

Reply:
If it gets past the point where it finds its attached shop, it _must_  be a Chest at that point. No reason to check again.
`.getSecondHalf()` will return null if the given block is not a Chest.